src/easylink/step.py

```diff
@@ -20,11 +21,13 @@ class Step(ABC):
 
     def __init__(
         self,
-        name: str,
+        step_name: str,
```

Comment:
This is the first time where we might use the same step or the same implementation multiple times in the same pipeline, so it is where explicit or implicit requirements around uniqueness of names become important. 
We need to maintain the idea of a "step name", perhaps poorly named, to give us the idea that this is something abstract that an implementation can implement ("step_3")
However, we also depend a lot on having a unique identifier for the node that plays some instance role in the pipeline schema ("step_3_loop_1").
We might need to do something a bit more elegant than this to solve the problem properly. Managing the interchange between different levels of the nested schema graph and the slots and names involved is what makes this a bit fussy to develop.

Reply:
I like "step name" FWIW